frontend/src/components/elements/PlotlyChart/CustomTheme.tsx

```diff
@@ -0,0 +1,640 @@
+            sequential: sequentialColorscaleDarkTheme,
+            sequentialminus: sequentialColorscaleLightTheme,
+          }),
+    },
+    // specifically for the ternary graph
+    ternary: {
+      gridcolor: getGray70(theme),
+      bgcolor: colors.bgColor,
+      title: {
+        font: {
+          family: genericFonts.bodyFont,
+          size: fontSizes.smPx,
+        },
+      },
+      color: getGray70(theme),
+      aaxis: {
+        gridcolor: getGray70(theme),
+        linecolor: getGray70(theme),
+        tickfont: {
+          family: genericFonts.bodyFont,
+          size: fontSizes.twoSmPx,
+        },
+      },
+      baxis: {
+        linecolor: getGray70(theme),
+        gridcolor: getGray70(theme),
+        tickfont: {
+          family: genericFonts.bodyFont,
+          size: fontSizes.twoSmPx,
+        },
+      },
+      caxis: {
+        linecolor: getGray70(theme),
+        gridcolor: getGray70(theme),
+        tickfont: {
+          family: genericFonts.bodyFont,
+          size: fontSizes.twoSmPx,
+        },
+      },
+    },
+  }
+
+  assign(layout, streamlitTheme)
+}
+
+/**
+ * Applies the colorscale, colors for unique graphs, and discrete coloring
+ * for in general through assigning properties in spec.data
+ * @param data - spec.data
+ */
+export function applyStreamlitThemeData(data: any): void {
+  const { colors } = useTheme()
+  applyColorscale(data)
+  applyUniqueGraphColorsData(data)
+  changeDiscreteColors(data)
+  data.forEach((entry: any) => {
+    if (entry.marker !== undefined) {
+      entry.marker.line = assign(entry.marker.line, {
+        width: 0,
+        color: colors.transparent,
+      })
+    }
+  })
+}
+
+/**
+ * This applies specific changes for spec.layout.template.data because
+ * spec.data does not fully catch these and neither does spec.layout.template.layout
+ * @param data - spec.layout.template.data
+ * @param theme - Theme from useTheme()
+ */
+export function applyStreamlitThemeTemplateData(
+  data: any,
+  theme: Theme
+): void {
+  if (data !== undefined) {
+    data.bar.forEach((entry: any) => {
+      if (entry.marker !== undefined && entry.marker.line !== undefined) {
+        entry.marker.line = assign(entry.marker.line, {
+          color: getGray30(theme),
+        })
+      }
+    })
+    const graphs = Object.values(data)
+    graphs.forEach((graph: any) => {
+      if (graph.colorbar !== undefined && graph.colorbar.ticks === "") {
+        // make tick values show
+        delete graph.colorbar.ticks
+      }
+    })
+  }
+}
+
+/**
+ * Applies the Streamlit theme by overriding properties in
+ * spec.data, spec.layout.template.data, and spec.layout.template.layout
+ * @param spec - spec
+ */
+export function applyStreamlitTheme(spec: any): void {
+  const theme: Theme = useTheme()
+  applyStreamlitThemeTemplateLayout(spec.layout.template.layout, theme)
+  applyStreamlitThemeTemplateData(spec.layout.template.data, theme)
+  applyStreamlitThemeData(spec.data)
```

Comment:
I think it would be good to add a try-catch around `applyStreamlitThemeData` since I can imagine a few problems when plotly applies changes to its spec. It is better to show the chart with slightly wrong colors in this case, then to have the full chart broken. The catch should just print a warning to the dev console.